app/services/sign_in/state_payload_jwt_encoder.rb

```diff
@@ -71,13 +78,18 @@ def state_payload
                                           client_id: client_config.client_id,
                                           code_challenge: remove_base64_padding(code_challenge),
                                           code: state_code,
-                                          client_state:)
+                                          client_state:,
+                                          scope:)
     end
 
     def state_code
       @state_code ||= SecureRandom.hex
     end
 
+    def sso_not_enabled_for_device_sso_scope?
+      scope == Constants::Auth::DEVICE_SSO && !client_config.device_sso_enabled?
+    end
```

Comment:
Total nitpick, so ignore if you disagree. I think it might be more clear if we refactored this a bit:

```ruby
    def sso_enabled_for_device_sso_scope?
      scope == Constants::Auth::DEVICE_SSO && client_config.device_sso_enabled?
    end
```

Then, update where it is called:

```ruby
    def validate_scope
      raise Errors::InvalidScope.new message: 'Scope is not valid for Client' unless sso_enabled_for_device_sso_scope?
    end
```

Reply:
the problem is this raises if scope is nil, regardless of the `client_config.device_sso_enabled?`

Reply:
We need it to raise only with one set of conditions:
- `scope` is 'device_sso'
- `client.config.device_sso_enabled?` is false

(but I def agree this conditional is convoluted, if you're able to think of how to make it more readable)